rust/lancedb/src/utils.rs

```diff
@@ -175,6 +176,25 @@ pub fn supported_vector_data_type(dtype: &DataType) -> bool {
     }
 }
 
+/// Note: this is temporary until we get a proper datatype conversion in Lance.
+pub fn string_to_datatype(s: &str) -> Option<DataType> {
```

Comment:
Ah, so we are using arrow types :).  I do think we will want this behavior documented more clearly at some point.  Doesn't have to be this PR.